src/mlpack/core/data/load_impl.hpp

```diff
@@ -164,8 +164,17 @@ bool Load(const std::string& filename,
 
   // We can't use the stream if the type is HDF5.
   bool success;
+  std::string print_status;
+
   if (loadType != arma::hdf5_binary)
-    success = matrix.load(stream, loadType);
+  {
+    if(loadType == arma::csv_ascii)
+      success = LoadCSVV(matrix, stream, print_status);
+    else
+      success = matrix.load(stream, loadType);
```

Comment:
Do we need an `else` condition here?

Reply:
Maybe, what's happening is the `if` condition is not specific to `csv_ascii`
`if (loadType != arma::hdf5_binary)`

Before our changes `matrix.load()` was kinda handling many formats here.
But we replaced it with a load function specific to `csv_ascii`.

So when some other format is being used it is failing the `load_save_test`.
Not sure if this is a proper fix just added that as a temporary patch.

Reply:
I still can not understand the issue, there are two lines here in your code 178, and 179. They should handle other cases, unless if I am missing something

Reply:
Also, the other thing that bothers me with this fix, is the use of the matrix itself to load the code. If we use bandicoot and it does not have the loading functionality then the user will not be able to load anything. Here, we are supposing that the users are using a matrix that has the load functionality inside as follows: ( just writing it down so we can back to it later)
`success = matrix.load(filename, loadType);`
I do not want to keep it like this, but maybe it is the best solution. We will see later.